client-java/src/main/java/uk/gov/gchq/palisade/client/download/Downloader.java

```diff
@@ -0,0 +1,246 @@
+         */
+        Resource getResource();
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Downloader.class);
+
+    private static final HttpClient HTTP_CLIENT = HttpClient.newBuilder().build();
+    private static final int HTTP_STATUS_OK = 200;
+    private static final int HTTP_STATUS_NOT_FOUND = 404;
+
+    private final DownloaderSetup setup;
+
+    private Downloader(final DownloaderSetup setup) {
+        assert setup != null : "Cannot create downloader without a setup!";
+        this.setup = setup;
+    }
+
+    /**
+     * Helper method to create a {@code Downloader} using a builder function
+     *
+     * @param func The builder function
+     * @return a newly created data request instance
+     */
+    @SuppressWarnings("java:S3242") // I REALLY want to use UnaryOperator here SonarQube!!!
+    static Downloader createDownloader(final UnaryOperator<DownloaderSetup.Builder> func) {
+        checkArgument(func);
+        return new Downloader(func.apply(DownloaderSetup.builder()).build());
+    }
+
+    @Override
+    public Optional<Object> findProperty(final String key) {
+        checkArgument(key);
+        return setup.getConfiguration().find(key);
+    }
+
+    /**
+     * Returns the downloadId associated with this downloader
+     *
+     * @return the downloadId associated with this downloader
+     */
+    UUID getDownloadId() {
+        return setup.getId();
+    }
+
+    @Override
+    public Resource getResource() {
+        return setup.getResource();
+    }
+
+    /**
+     * Start the download process
+     *
+     * @return a download result after successful completion
+     * @throws DownloaderException if any error occurs
+     */
+    @SuppressWarnings("java:S2221")
+    public DownloadResult start() {
+
+        LOGGER.debug("Downloader Started");
+
+        var mapper = setup.getObjectMapper();
+        var resource = setup.getResource();
+        var receiver = setup.getReceiver();
+
+        try {
+
+            var uri = Util.createUri(resource.getUrl(), setup.getConfiguration().getDataPath());
+
+            var requestBody = mapper
+                .writeValueAsString(createDataRequest(b -> b
+                    .token(resource.getToken())
+                    .leafResourceId(resource.getLeafResourceId())));
+
+            var httpResponse = sendRequest(requestBody, uri);
+
+            var statusCode = httpResponse.statusCode();
+
+            if (statusCode != HTTP_STATUS_OK) {
+                String msg;
+                if (statusCode == HTTP_STATUS_NOT_FOUND) {
+                    msg = String.format("Resource \"%s\" not found", resource.getLeafResourceId());
+                } else {
+                    msg = "Request to DataService failed";
+                }
+                throw new DownloaderException(msg, statusCode);
+            }
+
+            var recieverResult = processStream(httpResponse.body(), receiver, this);
```

Comment:
change spelling and update line 201 with updated var
 ```suggestion
            var receiverResult = processStream(httpResponse.body(), receiver, this);
```

Reply:
fixed